src/hotspot/share/runtime/deoptimization.hpp

```diff
@@ -342,7 +342,10 @@ class Deoptimization : AllStatic {
       return 0;
     }
   }
-  static int trap_request_index(int trap_request) {
+  // In RISC-V, when this method is inlined, the condition behaves
+  // incorrectly if the nmethod is compiled by JVMCI, causing the
+  // method to return a negative index which is not -1
+  static int RISCV64_ONLY(__attribute__ ((noinline))) trap_request_index(int trap_request) {
```

Comment:
I don't understand this change, because the semantics should be the same whether inline or not. What's the difference  in the generated code by the c/c++ compiler?

Reply:
Here is the assembly code without inlining:
```
  49b768:	57fd                	li	a5,-1
  49b76a:	00054363          	bltz	a0,49b770 <_ZN14Deoptimization18trap_request_indexEi+0xe>
  49b76e:	87aa                	mv	a5,a0
  49b770:	6422                	ld	s0,8(sp)
  49b772:	0007851b          	sext.w	a0,a5
```

and here is the code with inlining:
```
  7de6a4:	000a879b          	sext.w	a5,s5
  7de6a8:	000ad363          	bgez	s5,7de6ae <_ZN14Deoptimization19uncommon_trap_innerEP10JavaThreadi+0x2ea>
  7de6ac:	57fd                	li	a5,-1
  7de6ae:	757d                	lui	a0,0xfffff
```
(a1 was moved in s5).
I don't see any semantic difference, so I am not sure why it would change. However, I tried to build and test both versions several times and the issue only appeared when the method was inlined.

Reply:
Sorry for the delay, but I managed to remove the attribute. There was a value that was not sign extended, but it had to be despite being an int. For some reason, it seems that removing the inlining sign extended the value.